draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2718,6 +2718,12 @@ Host: www.example.org
    cache. See <xref target="security.considerations"/> for security
    considerations regarding message routing.
 </t>
+<t>
+   In particular, an origin server &MUST; reject a request whose target URI
+   has a scheme that implies it is to be served over a cryptographically
+   secured connection (such as "https") when the connection it was received
+   upon was not secured, unless the caveat above applies.
+</t>
```

Comment:
```suggestion
<t>
   Unless the caveat above applies, an origin server &MUST; reject or redirect
   a request if the target URI's minimum requirements for access to the identified
   resource are not being met by the request context.
   In particular, a request for an "https" resource is illegitimate unless it has been
   received over a connection that has been secured via a certificate
   valid for that target URI's origin, as defined by <xref target="https.uri"/>.
</t>
```

Reply:
I don't mind that approach, but:

* This phrasing is too obfuscated: "... the target URI's minimum requirements for access to the identified resource are not being met by the request context"

Suggestion: "... any scheme-specific requirements are not met."

* "illegitimate" isn't used anywhere else, and it's unclear what it means here. Suggest "MUST be refused" or "will be refused" if you want to avoid the reinforcing requirement.